website/src/models/group.js

```diff
@@ -168,6 +170,81 @@ schema.methods.isMember = function isGroupMember (user) {
   }
 };
 
+schema.methods.startQuest = async function startQuest (user) {
+  if (this.type !== 'party') throw new BadRequest('Must be a party to use this method');
+  if (!this.quest.key) throw new BadRequest('Party does not have a pending quest');
+  if (this.quest.active) throw new BadRequest('Quest is already active');
+
```

Comment:
Actually, I believe in the specification for http error codes it states that 400 error codes you depict user errors and 500 should depict fault in our part. So, we shouldn't give 500s if it is the user's fault.